app/src/main/java/ch/epfl/culturequest/database/Database.java

```diff
@@ -260,11 +261,18 @@ public static CompletableFuture<AtomicBoolean> updateScore(String uid, int newSc
         CompletableFuture<AtomicBoolean> future = new CompletableFuture<>();
         DatabaseReference scoreRef = databaseInstance.getReference("users/" + uid + "/score");
         scoreRef.setValue(newScore, (error, ref) -> {
-            if (error != null) {
-                future.complete(new AtomicBoolean(false));
-            } else {
-                future.complete(new AtomicBoolean(true));
-            }
+            future.complete(new AtomicBoolean(error == null));
+        });
+        return future;
+    }
+
+
+
+    public static CompletableFuture<AtomicBoolean> updateBadges(String uid, HashMap<String,Integer> newbadges) {
+        CompletableFuture<AtomicBoolean> future = new CompletableFuture<>();
+        DatabaseReference badgesRef = databaseInstance.getReference("users/" + uid + "/badges");
+        badgesRef.setValue(newbadges, (error, ref) -> {
```

Comment:
Rather than a set value, I suggest you use a transaction as I did with the list of likers or followers. It is better to guarantee that you update the newest version, in case the hasmap you have locally is not fully up to date.